docs/content/docs/deployment/elastic_scaling.md

```diff
@@ -0,0 +1,112 @@
+ 
+### Getting started
+
+If you just want to try out Reactive Mode, follow these instructions. They assume that you are deploying Flink on one machine.
+
+```bash
+
+# these instructions assume you are in the root directory of a Flink distribution.
+
+# Put Job into lib/ directory
+cp ./examples/streaming/TopSpeedWindowing.jar lib/
+# Submit Job in Reactive Mode
+./bin/standalone-job.sh start -Drestart-strategy=fixeddelay -Drestart-strategy.fixed-delay.attempts=100000 -Dscheduler-mode=reactive -j org.apache.flink.streaming.examples.windowing.TopSpeedWindowing
+# Start first TaskManager
+./bin/taskmanager.sh start
+```
+
+Let's quickly examine the used submission command:
+- `./bin/standalone-job.sh start` deploys Flink in [Application Mode]({{< ref "docs/deployment/overview" >}}#application-mode)
+- `-Drestart-strategy=fixeddelay` and `-Drestart-strategy.fixed-delay.attempts=100000` configure the job to restart on failure. This is needed for supporting scale-down.
+- `-Dscheduler-mode=reactive` enables Reactive Mode.
+- the last argument is passing the Job's main class name.
+
+You have now started a Flink job in Reactive Mode. The [web interface](http://localhost:8081) shows that the job is running on one TaskManager. If you want to scale up the job, simply add another TaskManager to the cluster:
+```bash
+# Start additional TaskManager
+./bin/taskmanager.sh start
+```
+
+To scale down, remove a TaskManager instance.
+```bash
+# Remove a TaskManager
+./bin/taskmanager.sh stop
+```
+
+### Usage
+
+#### Configuration
+
+To enable Reactive Mode, you need to configure `scheduler-mode` to `reactive`.
+
+The **parallelism of individual operators in a job will be determined by the scheduler**. It is not configurable. 
+
+The only way of influencing the parallelism is by setting a max parallelism for a operator (which will be respected by the scheduler). The maxParallelism is bounded by 2^15 (32768), which is the value that Reactive Mode uses if nothing else is configured. If there is no maxParallelism defined for an operator, `32768` will be used as a default.
+If you manually set a parallelism in your job for individual operators or the entire job, this setting will be ignored.
+
+Note that such a high maxParallelism might affect performance of the job, since more internal structures are needed to maintain [some internal structures](https://flink.apache.org/features/2017/07/04/flink-rescalable-state.html) of Flink.
+
+#### Recommendations
+
+- **Configure periodic checkpointing for stateful jobs**: Reactive mode restores from the latest completed checkpoint on a rescale event. If no periodic checkpointing is enabled, your program will loose its state.
```

Comment:
```suggestion
- **Configure periodic checkpointing**: Reactive mode restores from the latest completed checkpoint on a rescale event. If periodic checkpointing is not enabled, your program will loose its state.
```

In the end bascially every application is stateful. Don't need to let the user think about whether it is stateful or not.

Generally, this sounds very dangerous. Wondering if failing if no checkpointing is configured would be the better option. :shrug:

Reply:
What if speed is more important than correctness, and a user explicitly needs no checkpointing (say you are just pumping logs from one Kafka to another, and you can tolerate a few seconds of data loss?)
I'm concerned that we are too restrictive by failing. Also, reactive mode is an MVP feature.  If we get feedback that people run into this, then we can revisit.

Reply:
Sounds good.